src/multihash.rs

```diff
@@ -54,12 +54,41 @@ impl<const S: usize> Default for Multihash<S> {
     }
 }
 
+/// Returned from [`Multihash::new`]
+#[derive(Debug)]
+#[non_exhaustive]
+pub enum CreationError {
+    /// The digest exceeds the capacity of the multihash
+    DigestTooBig,
+}
+
+impl fmt::Display for CreationError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            CreationError::DigestTooBig => {
+                f.write_str("The digest exceeds the capacity of the multihash")
+            }
+        }
+    }
+}
+
+#[cfg(feature = "std")]
+impl std::error::Error for CreationError {}
+
 impl<const S: usize> Multihash<S> {
-    /// Wraps the digest in a multihash.
-    pub const fn wrap(code: u64, input_digest: &[u8]) -> Result<Self, Error> {
+    /// Wrap the digest in a multihash
+    /// ```rust
+    /// # use multihash::Multihash;
+    /// const MULTIHASH: Multihash<64> = match Multihash::new(0, &[]) {
+    ///     Ok(ok) => ok,
+    ///     Err(_) => panic!(),
+    /// };
+    /// ```
```

Comment:
```suggestion
    /// # Example usage in `const`
    ///
    /// ```rust
    /// # use multihash::Multihash;
    /// const MULTIHASH: Multihash<64> = match Multihash::new(0, &[]) {
    ///     Ok(ok) => ok,
    ///     Err(_) => panic!(),
    /// };
    /// ```
```